dockerfiles/base/scripts/base/commands/cmd_stop.sh

```diff
@@ -29,13 +29,23 @@ cmd_stop() {
   FORCE_STOP=false
   if [[ "$@" == *"--skip:graceful"* ]]; then
   	FORCE_STOP=true
+  elif local_repo; then
+    warning "Development mode [skip graceful stop]"
+    FORCE_STOP=true
   fi
 
   if server_is_booted $(get_server_container_id $CHE_CONTAINER_NAME); then 
     if [[ ${FORCE_STOP} = "false" ]]; then
       info "stop" "Stopping workspaces..."
-      if ! $(cmd_lifecycle action "graceful-stop" "$@" >> "${LOGS}" 2>&1 || false); then
-        error "We encountered an error -- see cli.log"
+      local GRACEFUL_STATUS_RESULT=0
+      cmd_lifecycle action "graceful-stop" "$@" >> "${LOGS}" 2>&1 || GRACEFUL_STATUS_RESULT=$?
+      # error on authentication (401 modulo 256 = 145)
+      if [[ ${GRACEFUL_STATUS_RESULT} -eq 145 ]]; then
+        error "Authentication failed on the system. Please provide --user and -password values or user --skip:graceful to bypass graceful stop."
+        return 2;
+      elif [[ ${GRACEFUL_STATUS_RESULT} -ne 0 ]]; then
+        error "We encountered an error -- see $CHE_HOST_CONFIG/cli.log. Graceful stop can be skipped with --skip:graceful"
```

Comment:
If you do it have it as (hint: `--skip:graceful` does not wait for workspace stop)